command/peering/export/export.go

```diff
@@ -0,0 +1,187 @@
+package export
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/cli"
+
+	"github.com/hashicorp/consul/agent"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/command/flags"
+)
+
+func New(ui cli.Ui) *cmd {
+	c := &cmd{UI: ui}
+	c.init()
+	return c
+}
+
+type cmd struct {
+	UI    cli.Ui
+	flags *flag.FlagSet
+	http  *flags.HTTPFlags
+	help  string
+
+	serviceName      string
+	serviceNamespace string
+	peerNames        string
+}
+
+func (c *cmd) init() {
+	//This function defines the flags
+
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+
+	c.flags.StringVar(&c.serviceName, "service", "", "(Required) Specify the name of the service you want to export.")
+	c.flags.StringVar(&c.serviceNamespace, "serviceNamespace", "", "(Optional) Specify the Namespace of the service, you want to export. Enterprise only feature")
```

Comment:
@DanStough we were thinking it might make sense to use the `namespace` flag inherited via `flags.Merge(c.flags, c.http.ClientFlags())` (and maybe the partition flag inherited the same way) instead of defining a new flag here. Thoughts?

Reply:
It doesn't look like we use that flag for anything else relevant to storing the `exported-services`, but it will automatically get added as a query parameter in the request: https://github.com/hashicorp/consul/blob/c046d1a4d870639227baff629ff304a1b72deede/api/api.go#L1019-L1021

No idea how Consul will handle that. You could test it out or use a dedicated field to be safe.